orchagent/portsorch.cpp

```diff
@@ -3851,12 +3930,43 @@ bool PortsOrch::isVlanMember(Port &vlan, Port &port)
     return true;
 }
 
-bool PortsOrch::addLag(string lag_alias)
+bool PortsOrch::addLag(string lag_alias, uint32_t spa_id, int32_t switch_id)
 {
     SWSS_LOG_ENTER();
 
+    vector<sai_attribute_t> lag_attrs;
+    string system_lag_alias = lag_alias;
+
+    if (gMySwitchType == "voq")
+    {
+        if (switch_id < 0)
+        {
+            // Local PortChannel. Allocate unique lag id from central CHASSIS_APP_DB
+            // Use the chassis wide unique system lag name.
+
+            // Get the local switch id and derive the system lag name.
+
+            switch_id = gVoqMySwitchId;
+            system_lag_alias = gMyHostName + "|" + gMyAsicName + "|" + lag_alias;
+
+            // Allocate unique lag id
+            spa_id = m_lagIdAllocator->lagIdAdd(system_lag_alias, 0);
```

Comment:
Currently Sonic in the teammgrd we don't set a unique oper key in the LACP BPDU, but we are in the process of adding this support (https://github.com/Azure/sonic-swss/pull/1660)

We could use this unique id allocation schema there as well. The change there would be to allocate the lagId in teammgrd when user creates a new LAG .. and here in orchagent it should give back the same id as it should be already existing ( created before in teammgrd and updated in SYSTEM LAG TABLE.)

Reply:
The need for this unique lag id allocation is to have unique lag id for all port channels across all the asics. So we let the id allocation be done in centralized location. As you mentioned, it will be interesting use case to use uniquely allocated id as key for LACP. This will work well for multi-asic and distributed chassis. For single asic, we need to set the id boundaries and load the script in local redis server. Yes, the lag id allocator gives the same id for a given portchannel, if it is already allocated,

Reply:
Yes in future we would need to extend this to teamd, so that we use this logic with the LUA script itself to generate a unique LACP key filled in the LACP BPDU.